bsp/boards/scum/radiotimer.c

```diff
@@ -0,0 +1,167 @@
+/**
+\brief SCuM-specific definition of the "radiotimer" bsp module.
+
+\author Tengfei Chang <tengfei.chang@inria.fr>, August 2016.
+*/
+
+#include "memory_map.h"
+#include "stdio.h"
+#include "stdint.h"
+#include "string.h"
+#include "radiotimer.h"
+#include "board.h"
+#include "bsp_timer.h"
+
+//=========================== define =======================================
+
+// ==== radio timer control bit
+
+#define RADIOTIMER_ENABLE                       0x01
+#define RADIOTIMER_INTERRUPT_ENABLE             0x02
+#define RADIOTIMER_COUNT_RESET                  0x04
+
+// ==== radio timer interruption flag
+
+#define RADIOTIMER_COMPARE0_INT                 0x0001
+#define RADIOTIMER_COMPARE1_INT                 0x0002
+#define RADIOTIMER_COMPARE2_INT                 0x0004
+#define RADIOTIMER_COMPARE3_INT                 0x0008
+#define RADIOTIMER_COMPARE4_INT                 0x0010
+#define RADIOTIMER_COMPARE5_INT                 0x0020
+#define RADIOTIMER_COMPARE6_INT                 0x0040
+#define RADIOTIMER_COMPARE7_INT                 0x0080
+#define RADIOTIMER_CAPTURE0_INT                 0x0100
+#define RADIOTIMER_CAPTURE1_INT                 0x0200
+#define RADIOTIMER_CAPTURE2_INT                 0x0400
+#define RADIOTIMER_CAPTURE3_INT                 0x0800
+#define RADIOTIMER_CAPTURE0_OVERFLOW_INT        0x1000
+#define RADIOTIMER_CAPTURE1_OVERFLOW_INT        0x2000
+#define RADIOTIMER_CAPTURE2_OVERFLOW_INT        0x4000
+#define RADIOTIMER_CAPTURE3_OVERFLOW_INT        0x8000
+
+//=========================== variables =======================================
+
+typedef struct {
+    radiotimer_compare_cbt    overflow_cb;
+    radiotimer_compare_cbt    compare_cb;
+    PORT_RADIOTIMER_WIDTH     interrupt_flag;
+} radiotimer_vars_t;
+
+radiotimer_vars_t radiotimer_vars;
+
+//=========================== prototypes ======================================
+
+//=========================== public ==========================================
+
+//===== admin
+
+void radiotimer_init() {
+    // clear local variables
+    memset(&radiotimer_vars,0,sizeof(radiotimer_vars_t));
+}
+
+void radiotimer_setOverflowCb(radiotimer_compare_cbt cb) {
+    radiotimer_vars.overflow_cb    = cb;
+}
+
+void radiotimer_setCompareCb(radiotimer_compare_cbt cb) {
+    radiotimer_vars.compare_cb     = cb;
+}
+
+void radiotimer_setStartFrameCb(radiotimer_capture_cbt cb) {
+    while(1);

```

Comment:
why `while(1)`? We took the VERY bad habbit to putting a `while(1)` to mean "I should never be here". But without JTAG, it just causes the board to halt without being able to debug. I would replace by a comment that says "should never be called".